Dockerfile

```diff
@@ -1,3 +1,8 @@
+# Docker image for checking integration across the main repo (allennlp) and
```

Comment:
I'm a bit confused about how the directory structure is set up (not sure how you can copy `allennlp` and `allennlp-semparse` as you do), but I'll trust you've done whatever magic is necessary in the TC build machine.

Reply:
Sorry for leaving that unclear! Added comment:

# We expect that the various repos are checked out under a directory matching
# their name, e.g. allennlp-semparse. A TeamCity build can be configured to do
# this by adding the checkout rule "+:.=>allennlp-semparse". These are found
# under "Version Control Settings" -> "VCS Roots" -> "Edit checkout rules".

I've also modified the other Dockerfile (and build) to have a more typical structure and not rely on this. (Since it's only installing packages for allennlp and allennlp-semparse.)

Reply:
Sorry for leaving that unclear! Added comment:

```
# We expect that the various repos are checked out under a directory matching
# their name, e.g. allennlp-semparse. A TeamCity build can be configured to do
# this by adding the checkout rule "+:.=>allennlp-semparse". These are found
# under "Version Control Settings" -> "VCS Roots" -> "Edit checkout rules".
```

I've also modified the other Dockerfile to adopt a more typical structure so that this isn't required for the common case. (Since we're only installing packages.)